### Leitura do arquivo .pkl

Leitura do arquivo onde foi salvo os previsores e as classes com seu devido pre processsamento


In [1]:
import pickle

In [2]:
with open('../../variaveis_preprocessadas/census.pkl', 'rb') as f:
    previsores_treinamento, classe_treinamento, previsores_teste, classe_teste = pickle.load(f)

### Utilização do algoritmo KNN

In [23]:
from sklearn.neighbors import KNeighborsClassifier

knn = KNeighborsClassifier(n_neighbors=5, p=2)
knn.fit(previsores_treinamento, classe_treinamento)
previsoes = knn.predict(previsores_teste)

### Avaliação dos resultados do algoritmo KNN

In [15]:
from sklearn.metrics import accuracy_score, confusion_matrix, classification_report

In [24]:
score = accuracy_score(classe_teste, previsoes)
score

0.8223132036847492

## Valores obtidos por parametros diferentes:

n_neighbours = 3; score = 0.8081  <br>
n_neighbours = 5; score = 0.8223  <br>
n_neighbours = 10; score = 0.8290 <br>

In [25]:
confusion_matrix(classe_teste, previsoes)

array([[3336,  357],
       [ 511,  681]], dtype=int64)

In [26]:
print(classification_report(classe_teste, previsoes))

              precision    recall  f1-score   support

       <=50K       0.87      0.90      0.88      3693
        >50K       0.66      0.57      0.61      1192

    accuracy                           0.82      4885
   macro avg       0.76      0.74      0.75      4885
weighted avg       0.82      0.82      0.82      4885

